src/utils/events.ts

```diff
@@ -0,0 +1,47 @@
+type Callback<Item> = Item extends { name: infer Name; type: infer Type }
```

Comment:
This is specific to StarkNet, shouldn't this be stored at `src/providers/starknet/events.ts`?